components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/dao/AbstractUserStoreDAO.java

```diff
@@ -0,0 +1,137 @@
+import org.wso2.carbon.identity.user.store.configuration.dto.UserStorePersistanceDTO;
+import org.wso2.carbon.identity.user.store.configuration.utils.IdentityUserStoreMgtException;
+import org.wso2.carbon.identity.user.store.configuration.utils.SecondaryUserStoreConfigurationUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is the abstract implementation of {@link UserStoreDAO}.
+ */
+public abstract class AbstractUserStoreDAO implements UserStoreDAO {
+
+    @Override
+    public void addUserStore(UserStoreDTO userStoreDTO) throws IdentityUserStoreMgtException {
+
+        UserStorePersistanceDTO userStorePersistanceDTO = getUserStorePersistanceDTO(userStoreDTO,
+                getUserStoreProperties(userStoreDTO));
+        doAddUserStore(userStorePersistanceDTO);
+    }
+
+    @Override
+    public void updateUserStore(UserStoreDTO userStoreDTO, boolean isStateChange) throws IdentityUserStoreMgtException {
+
+        if (isStateChange) {
+            userStoreDTO = getUserStoreProperty(userStoreDTO);
+        }
+        UserStorePersistanceDTO userStorePersistanceDTO = getUserStorePersistanceDTO(userStoreDTO,
+                getUserStoreProperties(userStoreDTO));
+        userStorePersistanceDTO.setUserStoreDTO(userStoreDTO);
+        doUpdateUserStore(userStorePersistanceDTO, isStateChange);
+    }
+
+    private UserStoreDTO getUserStoreProperty(UserStoreDTO userStoreDTO) throws IdentityUserStoreMgtException {
+
+        boolean newState = userStoreDTO.getDisabled();
+        UserStoreDTO userStoreDTOTemp = getUserStore(userStoreDTO.getDomainId());
+        if (userStoreDTOTemp != null) {
+            userStoreDTO = userStoreDTOTemp;
+            userStoreDTO.setDisabled(newState);
+            PropertyDTO[] propertyDTO = userStoreDTO.getProperties();
+            List<PropertyDTO> newPropertyDTO = new ArrayList<>();
+            for (PropertyDTO propertyDTOValue : propertyDTO) {
+                if (propertyDTOValue.getName().equals("Disabled")) {
+                    propertyDTOValue.setValue(String.valueOf(newState));
+                    newPropertyDTO.add(propertyDTOValue);
+                } else {
+                    newPropertyDTO.add(propertyDTOValue);
+                }
+            }
+            userStoreDTO.setProperties(newPropertyDTO.toArray(new PropertyDTO[newPropertyDTO.size()]));
+        }
+        return userStoreDTO;
+    }
+
+    @Override
+    public void updateUserStoreDomainName(String previousDomainName, UserStoreDTO userStoreDTO)
+            throws IdentityUserStoreMgtException {
+
+        UserStorePersistanceDTO userStorePersistanceDTO = getUserStorePersistanceDTO(userStoreDTO,
+                getUserStoreProperties(userStoreDTO));
+        doUpdateUserStoreDomainName(previousDomainName, userStorePersistanceDTO);
+    }
+
+    @Override
+    public UserStoreDTO getUserStore(String domain) throws IdentityUserStoreMgtException {
+
+        if (doGetUserStore(domain) != null) {
```

Comment:
Get assigned this to a variable doGetUserStore(domain). Otherwise, it executes twise